modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -267,14 +325,45 @@ public GridMetricManager(GridKernalContext ctx) {
      */
     public MetricRegistry registry(String name) {
         return registries.computeIfAbsent(name, n -> {
-            MetricRegistry mreg = new MetricRegistry(name, log);
+            MetricRegistry mreg = new MetricRegistry(name,
```

Comment:
That is why I suggested separating `MetricRegistry` from some helper class for metrics creation. It's not registry responsibility for getting config from metastorage. But ok. May be split this later.